Break out commonly used functions in tumor cell validation

allyhawkins opened the pull request:
### Purpose/implementation Section

#### Please link to the GitHub issue that this pull request addresses.

Preparation for #480 


#### What is the goal of this pull request?

As I was working on #480, I realized there are a few things that we are going to want to do in all of the notebooks that validate tumor cells and it might make sense to take that code out of the actual exploratory notebooks and put into functions that can be sourced in. I also thought it might be helpful to get these changes done separately from adding a whole new notebook so I'm doing this first. 

#### Briefly describe the general approach you took to achieve this goal.

I took the heatmap functions that I created in #500 and added them to a new script that lives in `scripts/utils`. I also added in two functions for creating a classification data frame and a marker gene data frame. The classification df has one row per cell and contains all the classifications from all methods we used. The marker gene data frame is an expanded version with one row per marker gene per cell. 

I then updated the existing notebook to use these functions and removed the code from that notebook. 

I made a few additional modifications based on things I was working on for #480:

- For `CopyKAT` I now am including the mean cnv detection for all chromosomes along with the predictions in the predictions output. That way we can create plots with it when we read in the `CopyKAT` output. 
- For `InferCNV` I am including the `scaled_mean_proportion` along with the predictions output. Again, this is so we can use this information in the plots used for validation. 


#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes 


### Results

No results right now as this is just reorganizing some existing code. 

### Author checklists

#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [ ] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

jashapiro commented on analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R:
```diff
@@ -0,0 +1,200 @@
+    # get rid of excess columns
+    dplyr::select(barcodes, UMAP1, UMAP2) |> 
+    # add in marker gene classifications
+    dplyr::left_join(marker_gene_results_df) |> 
+    # add in copykat results 
+    dplyr::left_join(copykat_results_df) |> 
+    # add in infercnv results and metadata
+    dplyr::left_join(infercnv_results_df) |> 
+    dplyr::left_join(infercnv_metadata_df) |> 
+    # gene set scores 
+    dplyr::left_join(geneset_scores_df)
```
add in `by = ` statements?
```suggestion
    dplyr::left_join(marker_gene_results_df) |> 
    # add in copykat results 
    dplyr::left_join(copykat_results_df) |> 
    # add in infercnv results and metadata
    dplyr::left_join(infercnv_results_df) |> 
    dplyr::left_join(infercnv_metadata_df) |> 
    # gene set scores 
    dplyr::left_join(geneset_scores_df)
```

jashapiro commented on analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R:
```diff
@@ -0,0 +1,200 @@
+# this function adds a row annotation since cells are rows 
+plot_cnv_heatmap <- function(
+    df,
+    cnv_col, 
```
Maybe something a bit more descriptive? Probably worth a comment too.
```suggestion
    cnv_col_prefix, 
```

jashapiro commented on analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R:
```diff
@@ -0,0 +1,200 @@
+  
+  # do a little bit of column reordering before converting to a matrix
+  cnv_df <- cnv_df |> 
+    dplyr::relocate(chr10, .before = chr11) |> 
+    dplyr::relocate(chr19, .before = chr20) |> 
+    dplyr::relocate(chr21, .before = chr22) |> 
```
I'm not sure why these specific columns are mis-ordered, but maybe we should set the full order for generality in case other mis-ordering pops up?
```suggestion
    dplyr::select(paste0("chr", 1:22)) |>
```

jashapiro reviewed (commented):
This looks good overall, but as long as you are pulling these chunks out into functions, I thought there were a few places that a bit of tidying might be worth doing. Mostly just removing extra messages from reading files and joins.

allyhawkins commented:
@jashapiro I believe I addressed all your clean up suggestions. I also updated the notebook to hide code so the html is much cleaner to look at. 

And then I addressed a few minor spelling errors found in #525.

jashapiro reviewed (approved):
LGTM!
